nutils/sample.py

```diff
@@ -1060,7 +1060,7 @@ def __init__(self, integrand: function.Array, sample: Sample) -> None:
         super().__init__(shape=integrand.shape, dtype=float if integrand.dtype in (bool, int) else integrand.dtype, spaces=integrand.spaces - frozenset(sample.spaces), arguments=integrand.arguments)
 
     def lower(self, args: function.LowerArgs) -> evaluable.Array:
-        ielem = evaluable.loop_index('_sample_' + '_'.join(self._sample.spaces), self._sample.nelems)
+        ielem = evaluable.loop_index(f'_sample_{len(args.args)}', self._sample.nelems)
```

Comment:
Can you explain this change?

Reply:
Without this change nested integrals over the same space would get the same loop index.

Reply:
Ah right, and now the index name is the nesting depth and therefore (locally) unique. Got it.